dask/array/routines.py

```diff
@@ -458,7 +458,7 @@ def gradient(f, *varargs, **kwargs):
         f.map_overlap(
             _gradient_kernel,
             dtype=f.dtype,
-            depth={j: 1 if j == ax else 0 for j in range(f.ndim)},
+            depth=1,
```

Comment:
This has a performance impact, though, right?

Reply:
The overhead of map_overlap increases pretty aggressively with the number of non-zero dimensions.

Reply:
Sounds like we are prioritizing the wrong thing here. Would argue that usability comes before performance.

It's pretty common for users to take the gradient on multiple dimensions and combine the results into some computation (e.g. gradient magnitude or gradient angle(s)). Without this fix, none of these are possible. IOW infinite time compared to somewhat longer than we would like.

This doesn't mean performance is irrelevant. Just that it will need to be addressed once issue ( https://github.com/dask/dask/issues/3634 ) is resolved. Currently the solution there is not obvious to me. A bunch of weird things seem to be happening.